modules/webgl/src/classes/texture.js

```diff
@@ -195,14 +195,23 @@ export default class Texture extends Resource {
         type: this.type,
         dataFormat: this.dataFormat,
         border: this.border,
-        mipmaps: false
+        mipmaps: this.mipmaps
       });
     }
     return this;
   }
 
   // Call to regenerate mipmaps after modifying texture(s)
   generateMipmap(params = {}) {
+    if (!this.mipmaps) {
+      if (this._isNPOT()) {
+        log.warn(`texture: ${this} is Non-Power-Of-Two, disabling mipmaping`)();
+      } else {
+        log.warn('mipmaps is not enabled.');
+      }
+      return this;
+    }
```

Comment:
It doesn't make sense to me to consider the mipmapping immutable like this. If I don't want to generate mipmaps immediate on initialization, but want to call this method later, there's no reason it should be blocked by the API. Something like this would make more sense to me:

```js
if (this._isNPOT()) {
  log.warn(`texture: ${this} is Non-Power-Of-Two, disabling mipmaping`)();
  return this;
}

this.mipmaps = true;
```

Reply:
👍